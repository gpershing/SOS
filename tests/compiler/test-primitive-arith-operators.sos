(* arithmetic operators on primitives int, float *)

int add(int x, int y) = x + y
int sub(int x, int y) = x - y
int mult(int x, int y) = x * y
int div(int x, int y) = x/y
int mod(int x, int y) = x%y

float add(float x, float y) = x + y 
float sub(float x, float y) = x - y 
float mult(float x, float y) = x * y 
float div(float x, float y) = x/y 

int x = add(1,2)
print(x) //3
x = sub(1,2)
print(x) //-1
x = mult(1,2)
print(x) //2
x = div(1,2)
print(x) //0
x = mod(10,8)
print(x) //2

float y = add(1.1, 2.1)
printf(y) //3.2
y = sub(2.1, 1.1)
printf(y) //1.0
y = mult(1.1, 2.1)
printf(y) //2.31
y = div(2.1, 1.1)
printf(y) //1.9090...
