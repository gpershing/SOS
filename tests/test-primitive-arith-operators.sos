/* arithmetic operators on primitives int, float */

add: (x: int, y: int) -> int = x + y
sub: (x: int, y: int) -> int = x - y
mult: (x: int, y: int) -> int = x * y
div: (x: int, y: int) -> int = x/y
mod: (x: int, y: int) -> int = x%y

addf: (x: float, y: float) -> float = x + y 
subf: (x: float, y: float) -> float = x - y 
multf:(x: float, y: float) -> float = x * y 
divf: (x: float, y: float) -> float = x/y 

x: int = add(1,2)
print(x) //3
x = sub(1,2)
print(x) //-1
x = mult(1,2)
print(x) //2
x = div(1,2)
print(x) //0
x = mod(10,8)
print(x) //2

y: float = addf(1.1, 2.1)
printf(y) //3.2
y = subf(2.1, 1.1)
printf(y) //1.0
y = multf(1.1, 2.1)
printf(y) //2.31
y = divf(2.1, 1.1)
printf(y) //1.9090...
