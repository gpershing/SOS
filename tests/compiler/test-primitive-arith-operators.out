3
-1
2
0
2
3.2
1.0
2.31
1.9090
